### ISSUE TITLE
Modularize react-scaffolder

### ISSUE THREAD

**Issue by rajikaimal** [99xt/react-scaffolder#23]
This is related to future work. Objective is to create an entire eco system of clients depending on a core set of APIs provided by react-scaffolder.

According to the proposed architecture core APIs will moved into a separate package. Probably `react-scaffolder-core`.

Core will be consumed by client application. Web client, CLI, Electron application, opening up numerous possibilities rather than sticking into a CLI based solution.

Proposed architecture: 

<img src="http://i.imgur.com/ev03Gqg.png" width="100%"/>

**Comment by shapkarin** [comment]
Maybe create a VS Code package?

**Comment by rajikaimal** [comment]
That's a good suggestion too. Great if you can provide any other inputs.

**Comment by shapkarin** [comment]
OK. Only after I make some from [that list](https://github.com/99xt/react-scaffolder/issues/53).
